src/database/memory.rs

```diff
@@ -512,13 +513,29 @@ macro_rules! populate_test_db {
         };
 
         let txid = tx.txid();
-        let confirmation_time = tx_meta
+        // Set Confirmation time only if current height is provided. Else None.
+        let confirmation_time = if let Some(height) = current_height {
+            tx_meta
             .min_confirmations
             .and_then(|v| if v == 0 { None } else { Some(v) })
             .map(|conf| $crate::BlockTime {
-                height: current_height.unwrap().checked_sub(conf as u32).unwrap() + 1,
+                height: height.checked_sub(conf as u32).unwrap() + 1,
                 timestamp: 0,
-            });
+            })
+        } else {
+            None
+        };
+
+        // Set the database sync height if current height is provided
+        if let Some(height) = current_height {
+            let sync_time = SyncTime {
+                block_time: BlockTime {
+                    height,
+                    timestamp: 0
+                }
+            };
+            db.set_sync_time(sync_time).unwrap();
```

Comment:
I think here you should only keep the most recent sync time, so instead of just setting it you should first check if something is present and compare the two

Reply:
It should.. Done.,.